packages/design-system/src/components/Pagination/Pagination.tsx

```diff
@@ -0,0 +1,187 @@
+import Ellipses from './Ellipses'
+import Page from './Page'
+import React, {useState} from 'react'
+
+const overflow = 1;
+const maxVisiblePages = 7;
+
+function paginationBuilder(currentPage, totalPages) {
```

Comment:
[fix] there are a few scenarios where the pagination component is only populating 5 slots instead of 7. 
<img width="511" alt="Screen Shot 2021-08-19 at 4 56 09 PM" src="https://user-images.githubusercontent.com/33579665/130154697-097ba57c-f77d-4179-b37b-dfc3e7a347f0.png">

I tried some different conditionals in this functions and came up with this to replace the conditionals currently here:
```js
// The number of slots needing to be calculated - so total - 2 since we now the first slow will always be 1 and the last will always be total pages
const availableSlots = maxVisiblePages - 2;
/**
   * if the current page is < 5, add 1 - 5 pages
   */
  if (currentPage < availableSlots) {
    start = 1;
    end = availableSlots;
  }

  /**
   * if current page is total pages - 1, make sure start begins one earlier
   */
  if (currentPage === totalPages - 2) {
    start -= 1;
  }

  /**
   * if end page is two from the end, make sure the last page shows instead of ellipsis
   */
  if (end === totalPages - 2) {
    end += 1;
  }

  /**
   * If `end` > `totalPages`, add last pages to `range[]`
   */
  if (end >= totalPages) {
    start = totalPages - (availableSlots - 1); 
    end = totalPages;
  }

  /**
   * If `totalPages` is 5 or fewer, all pages added to `range[]`
   */
  if (totalPages <= maxVisiblePages) {
    start = 1;
    end = totalPages;
  }
```

Reply:
Thanks for wiring up this logic!